sql/analyze_test.go

```diff
@@ -147,6 +147,8 @@ func TestSimplifyExpr(t *testing.T) {
 
 		{`a IN (1, 1)`, `a IN (1)`},
 		{`a IN (2, 3, 1)`, `a IN (1, 2, 3)`},
+		{`a IN (1, true)`, `true`},
```

Comment:
just read the comment on `typeCheckTuple`; I guess producing an error from this input will come later.